programs/merkle-distributor/src/instructions/claim_and_stake/new_claim_and_stake.rs

```diff
@@ -0,0 +1,212 @@
+
+    /// SPL [Token] program.
+    pub token_program: Program<'info, Token>,
+
+    /// The [System] program.
+    pub system_program: Program<'info, System>,
+
+    /// Voter program
+    pub voter_program: Program<'info, Voter>,
+
+    /// CHECK: Locker
+    #[account(mut)]
+    pub locker: Box<Account<'info, Locker>>,
+
+    /// CHECK: escrow
+    #[account(mut)]
+    pub escrow: Box<Account<'info, Escrow>>,
+
+    /// CHECK: escrow_tokens
+    #[account(mut)]
+    pub escrow_tokens: UncheckedAccount<'info>,
+}
+
+/// Initializes a new claim from the [MerkleDistributor].
+/// 1. Increments num_nodes_claimed by 1
+/// 2. Initializes claim_status
+/// 3. Transfers claim_status.unlocked_amount to the claimant
+/// 4. Increments total_amount_claimed by claim_status.unlocked_amount
+/// CHECK:
+///     1. The claim window has not expired and the distributor has not been clawed back
+///     2. The claimant is the owner of the to account
+///     3. Num nodes claimed is less than max_num_nodes
+///     4. The merkle proof is valid
+#[allow(clippy::result_large_err)]
+pub fn handle_new_claim_and_stake(
+    ctx: Context<NewClaimAndStake>,
+    amount_unlocked: u64,
+    amount_locked: u64,
+    proof: Vec<[u8; 32]>,
+) -> Result<()> {
+    let mut distributor = ctx.accounts.distributor.load_mut()?;
+
+    // check operator
+    distributor.validate_claim_and_stake(&ctx.accounts.operator)?;
+
+    let escrow = &ctx.accounts.escrow;
+    require!(escrow.is_max_lock, ErrorCode::EscrowIsNotMaxLock);
+
+    let activation_handler = distributor.get_activation_handler()?;
+    activation_handler.validate_claim()?;
+
+    distributor.num_nodes_claimed = distributor
+        .num_nodes_claimed
+        .checked_add(1)
+        .ok_or(ErrorCode::ArithmeticError)?;
+
+    require!(
+        distributor.num_nodes_claimed <= distributor.max_num_nodes,
+        ErrorCode::MaxNodesExceeded
+    );
+
+    let claimant_account = &ctx.accounts.claimant;
+
+    // Verify the merkle proof.
+    let node = hashv(&[
+        &claimant_account.key().to_bytes(),
+        &amount_unlocked.to_le_bytes(),
+        &amount_locked.to_le_bytes(),
+    ]);
+
+    let node = hashv(&[LEAF_PREFIX, &node.to_bytes()]);
+
+    require!(
+        verify(proof, distributor.root, node.to_bytes()),
+        ErrorCode::InvalidProof
+    );
+
+    let claim_status = &mut ctx.accounts.claim_status;
+
+    // Seed initial values
+    claim_status.distributor = ctx.accounts.distributor.key();
+    claim_status.claimant = claimant_account.key();
+    claim_status.locked_amount = amount_locked;
+    claim_status.unlocked_amount = amount_unlocked;
+    claim_status.locked_amount_withdrawn = 0;
+    claim_status.closable = distributor.closable();
+    claim_status.admin = distributor.admin;
+
+    let bonus =
+        distributor.get_bonus_for_a_claimaint(claim_status.unlocked_amount, &activation_handler)?;
+    let amount_with_bonus = claim_status.unlocked_amount.safe_add(bonus)?;
```

Comment:
The bonus was not appended to `claim_status.unlocked_amount`

Reply:
fixed